File: python/servidor.py
import socket

def servidor(host='127.0.0.1', port=12345):
    # Crear el socket (IPv4, TCP)
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as servidor_socket:
        servidor_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
        servidor_socket.bind((host, port))            # Vincular IP/puerto
        servidor_socket.listen(5)                     # Escuchar hasta 5 en cola
        print(f"Servidor escuchando en {host}:{port} ...")

        while True:
            # Aceptar una conexión entrante (bloquea hasta que llegue un cliente)
            cliente_socket, direccion_cliente = servidor_socket.accept()
            with cliente_socket:
                print(f"Conexión establecida con: {direccion_cliente}")

                # Enviar un mensaje al cliente
                mensaje = "¡Hola desde el servidor!"
                cliente_socket.sendall(mensaje.encode('utf-8'))

                # Recibir datos del cliente (máx 1024 bytes)
                datos = cliente_socket.recv(1024)
                if not datos:
                    print("Cliente cerró la conexión.")
                    continue

                print(f"Mensaje del cliente: {datos.decode('utf-8')}")

if __name__ == "__main__":
    servidor()

File: java/servidor.java
import java.io.*;
import java.net.*;

public class servidor {
    public static void main(String[] args) {
        int puerto = 12345;

        try (ServerSocket servidor = new ServerSocket(puerto)) {
            System.out.println("Servidor escuchando en el puerto " + puerto + "...");

            // Espera una conexión de un cliente
            Socket socketCliente = servidor.accept();
            System.out.println("Cliente conectado desde " + socketCliente.getInetAddress());

            // Flujo de salida: enviar mensaje al cliente
            PrintWriter out = new PrintWriter(socketCliente.getOutputStream(), true);
            out.println("¡Hola desde el servidor!");

            // Flujo de entrada: recibir mensaje del cliente
            BufferedReader in = new BufferedReader(new InputStreamReader(socketCliente.getInputStream()));
            String mensajeCliente = in.readLine();
            System.out.println("Mensaje del cliente: " + mensajeCliente);

            // Cerrar recursos
            socketCliente.close();

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
